### PR TITLE
Validate Saved Query Names

### LINKED ISSUE #180: Add Saved Query Name Validation
### Housekeeping

- [X] I am a maintainer of dbt-semantic-interfaces

### Short description

Currently we use the [UniqueAndValidNameRule](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/4f6cc55665e5fbe755dfaf8f9bc942020ef15ee5/dbt_semantic_interfaces/validations/unique_valid_name.py#L55) to validate that the names of semantic models, dimensions, measures, entities, and metrics are valid (and where necessary also unique). Saved queries are not currently part of this. We need to add saved queries to be part of this validation

### Acceptance criteria

The [UniqueAndValidNameRule](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/4f6cc55665e5fbe755dfaf8f9bc942020ef15ee5/dbt_semantic_interfaces/validations/unique_valid_name.py#L55) validates that saved query names are
- valid
- unique

### Impact to Other Teams

This change _shouldn't_ impact any teams. Saved queries were added in 0.3.0 of DSI which is part of dbt-core 1.7.0rc1. So some open source people on dbt-core 1.7.0rc1 might experience breakages if their saved query names are in conflict with the change. Given that 1.7.0rc1 has been out for 2.5 business days, I don't believe this is a major concern. On the MetricFlow side validations are generally considered _upstream_ and thus shouldn't affect how MetricFlow itself operates.

### Will backports be required?

0.3.latest because that is when saved queries were introduced.

### Context

_No response_

### PR DESCRIPTION
**Pull request by QMalcolm**
Resolves #180

### Description

In #148 which went out in [0.3.0](https://github.com/dbt-labs/dbt-semantic-interfaces/releases/tag/v0.3.0), we forgot to add a validation for the saved query name. This PR adds that validation.

We plan on backporting this to 0.3.latest, which would break people who have saved queries defined with invalid name. Luckily the main thing that uses DSI is dbt-core, and dbt-core's latest production version (1.6.5) uses DSI 0.2.latest. DSI 0.3.0 is being used by dbt-core 1.7.0r1, but not many people are using that yet. Thus it's unlikely that many people have defined saved queries, and only a subset of those who have will have defined invalid names.

### Checklist

- [X] I have read [the contributing guide](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [X] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [X] This PR includes tests, or tests are not required/relevant for this PR
- [X] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: dbt_semantic_interfaces/type_enums/node_type.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from dbt_semantic_interfaces.enum_extension import ExtendedEnum
+
+
+class NodeType(ExtendedEnum):
```
Comment (plypaul) [inline dbt_semantic_interfaces/type_enums/node_type.py:4]: Nit: There are a few different nodes in DSI / MF - can you name this more specific?

**Review by plypaul — APPROVED** [review]
LGTM aside from nit.

File: dbt_semantic_interfaces/type_enums/node_type.py (hunk 1 above)
Comment (tlento) [reply dbt_semantic_interfaces/type_enums/node_type.py:4]: +1. Maybe call this SemanticManifestNodeType or something?

Comment (QMalcolm) [reply dbt_semantic_interfaces/type_enums/node_type.py]: Done! 🙂

**Pull request merged**